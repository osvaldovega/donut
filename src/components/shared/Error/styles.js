import { StyleSheet } from 'react-native';
import Colors from '../../../constants/Colors';

const styles = StyleSheet.create({
  wrapper: {
    width: '100%',
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    flex: 1,
  },
  content: {
    fontSize: 20,
    color: Colors.pink,
    fontFamily: 'montserrat',
    fontWeight: '600',
    padding: 10,
  },
});

export default styles;
