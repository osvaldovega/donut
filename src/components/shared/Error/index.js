import React from 'react';
import { View, Text } from 'react-native';
import styles from './styles';

const Error = () => (
  <View style={[styles.wrapper]}>
    <Text style={[styles.content]}>We had a problem proccesing your request</Text>
    <Text style={[styles.content]}>Please check again later!!!</Text>
  </View>
);

export default Error;
