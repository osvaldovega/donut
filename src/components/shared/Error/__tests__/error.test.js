import React from 'react';
import renderer from 'react-test-renderer';
import Error from '../index';

describe('<Error />', () => {

  const setup = () => renderer.create(<Error />);

  it('renders correctly', () => {
    const tree = setup().toJSON();
    expect(tree).toMatchSnapshot();
  });
});
