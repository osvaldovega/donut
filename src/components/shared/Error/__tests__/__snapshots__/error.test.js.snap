// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ff7675",
          "fontFamily": "montserrat",
          "fontSize": 20,
          "fontWeight": "600",
          "padding": 10,
        },
      ]
    }
  >
    We had a problem proccesing your request
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#ff7675",
          "fontFamily": "montserrat",
          "fontSize": 20,
          "fontWeight": "600",
          "padding": 10,
        },
      ]
    }
  >
    Please check again later!!!
  </Text>
</View>
`;
